mediafile.py

```diff
@@ -131,15 +135,38 @@ def mutagen_call(action, path, func, *args, **kwargs):
     except mutagen.MutagenError as exc:
         log.debug(u'%s failed: %s', action, six.text_type(exc))
         raise UnreadableFileError(path, six.text_type(exc))
+    except UnreadableFileError:
+        # reraise our errors without changes
+        raise
     except Exception as exc:
         # Isolate bugs in Mutagen.
         log.debug(u'%s', traceback.format_exc())
         log.error(u'uncaught Mutagen exception in %s: %s', action, exc)
         raise MutagenError(path, exc)
 
 
+def loadfile(method=True, writable=False, create=False):
+    def wrap(func):
+        f = mutagen._util.loadfile(method, writable, create)(func)
+
+        def wrapper(*args, **kwargs):
+            return mutagen_call('loadfile', '', f, *args, **kwargs)
+        return wrapper
+    return wrap
+
 # Utility.
 
+
+def _update_filething(filething):
```

Comment:
The problem is I'm not exactly sure why and where exactly closing happens.
Debugging this issue was guessing game.
`mutagen` was closing my files (only in case of local files, wtf) after reading and I couldn't find any `.close()` call in whole mutagen code nor using debugger on own proxy file-like.
Anyway - this allows to reopen files in `r+` when saving when opened as `r` on loading.
Proxy file-like weren't closed at all..

I'm not sure what could I write in this docstring what's not written in comment anyway.

Reply:
I found the reason!
`mutagen._util._openfile` [closes files it opened](https://github.com/quodlibet/mutagen/blob/30f373fa6d56e1afa17d48a0c6f3e111267fbd32/mutagen/_util.py#L273) just after [function needing file finishes](https://github.com/quodlibet/mutagen/blob/30f373fa6d56e1afa17d48a0c6f3e111267fbd32/mutagen/_util.py#L155).